### PR TITLE
approvider -> app-provider

### PR DESCRIPTION
**Pull request by butonic**
This PR aligns the `app-provider` service name with the `app-registry` and the other `auth-*` and `storage-*` services.
- [ ] the individual services example needs an update https://github.com/owncloud/ocis/pull/3665#pullrequestreview-961108347

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -1721,7 +1721,7 @@ def ocisServer(storage, accounts_hash_difficulty = 4, volumes = [], depends_on =
             "SHARING_USER_OWNCLOUDSQL_DB_NAME": "owncloud",
             # General oCIS config
             # OCIS_RUN_EXTENSIONS specifies to start all extensions except glauth, idp and accounts. These are replaced by external services
-            "OCIS_RUN_EXTENSIONS": "settings,storage-metadata,graph,graph-explorer,ocs,store,thumbnails,web,webdav,frontend,gateway,user,group,auth-basic,auth-bearer,auth-machine,storage-users,storage-shares,storage-publiclink,appprovider,sharing,proxy,nats,ocdav",
+            "OCIS_RUN_EXTENSIONS": "settings,storage-metadata,graph,graph-explorer,ocs,store,thumbnails,web,webdav,frontend,gateway,user,group,auth-basic,auth-bearer,auth-machine,storage-users,storage-shares,storage-publiclink,app-provider,sharing,proxy,nats,ocdav",
```
Comment (butonic) [inline .drone.star:1724]: @wkloucek this is missing the app-registry, isn't it?

File: .vscode/launch.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
                 // metadata storage
                 "METADATA_USER_ID": "some-metadata-user-id"
                 // OCIS_RUN_EXTENSIONS allows to start a subset of extensions even in the supervised mode
-                //"OCIS_RUN_EXTENSIONS": "settings,storage-metadata,glauth,graph,graph-explorer,idp,ocs,store,thumbnails,web,webdav,storage-frontend,storage-gateway,storage-userprovider,storage-groupprovider,storage-authbasic,storage-authbearer,storage-authmachine,storage-users,storage-shares,storage-public-link,storage-appprovider,storage-sharing,accounts,proxy,ocdav",
+                //"OCIS_RUN_EXTENSIONS": "settings,storage-metadata,glauth,graph,graph-explorer,idp,ocs,store,thumbnails,web,webdav,storage-frontend,storage-gateway,storage-userprovider,storage-groupprovider,storage-authbasic,storage-authbearer,storage-authmachine,storage-users,storage-shares,storage-public-link,app-provider,storage-sharing,accounts,proxy,ocdav",
```
Comment (butonic) [inline .vscode/launch.json:44]: @wkloucek this is missing the app-registry, isn't it? Furthermore some service have the wrong name, eg `storage-public-link`

File: deployments/examples/oc10_ocis_parallel/docker-compose.yml [hunk 3]
```diff
@@ -108,7 +108,7 @@ services:
       OCIS_STORAGE_READ_ONLY: "false" # TODO: conflict with OWNCLOUDSQL -> https://github.com/owncloud/ocis/issues/2303
       # General oCIS config
       # OCIS_RUN_EXTENSIONS specifies to start all extensions except glauth, idp and accounts. These are replaced by external services
-      OCIS_RUN_EXTENSIONS: settings,storage-metadata,graph,graph-explorer,ocs,store,thumbnails,web,webdav,storage-frontend,storage-gateway,storage-userprovider,storage-groupprovider,storage-authbasic,storage-authbearer,storage-authmachine,storage-users,storage-shares,storage-public-link,storage-appprovider,storage-sharing,proxy,nats
```
Comment (butonic) [inline deployments/examples/oc10_ocis_parallel/docker-compose.yml:111]: @wkloucek again `app-registry` missing?

File: deployments/examples/ocis_individual_services/docker-compose.yml [hunk 4]
```diff
@@ -664,7 +664,8 @@ services:
       STORAGE_FRONTEND_LOG_COLOR: "${OCIS_LOG_COLOR:-false}"
       STORAGE_FRONTEND_LOG_PRETTY: "${OCIS_LOG_PRETTY:-false}"
 
-      STORAGE_FRONTEND_APPPROVIDER_INSECURE: "true"
+      # FIXME this now lives in a dedicated service
+      APP_PROVIDER_WOPI_INSECURE: "true"
```
Comment (butonic) [inline deployments/examples/ocis_individual_services/docker-compose.yml:668]: I'd like to defer this to a subsequent PR

File: extensions/app-provider/pkg/config/config.go [hunk 5]
```diff
@@ -68,7 +68,7 @@ type WOPIDriver struct {
 	AppInternalURL string `yaml:"app_internal_url" env:"APP_PROVIDER_WOPI_APP_INTERNAL_URL" desc:"internal url to the app, eg in your DMZ"`
 	AppName        string `yaml:"app_name" env:"APP_PROVIDER_WOPI_APP_NAME" desc:"human readable app name"`
 	AppURL         string `yaml:"app_url" env:"APP_PROVIDER_WOPI_APP_URL" desc:"url for end users to access the app"`
-	Insecure       bool   `yaml:"insecure" env:"APP_PROVIDER_WOPI_INSECURE" desc:"allow insecure connections to the app"`
+	Insecure       bool   `yaml:"insecure" env:"OCIS_INSECURE;APP_PROVIDER_WOPI_INSECURE" desc:"allow insecure connections to the app"`
```
Comment (butonic) [inline extensions/app-provider/pkg/config/config.go:71]: @wkloucek or was this left out intentionally?

File: ocis-pkg/config/config.go [hunk 6]
```diff
@@ -73,7 +73,7 @@ type Config struct {
 	Runtime           Runtime              `yaml:"runtime"`
 
 	Accounts          *accounts.Config        `yaml:"accounts"`
-	AppProvider       *appprovider.Config     `yaml:"appprovider"`
+	AppProvider       *appProvider.Config     `yaml:"app-provider"`
```
Comment (butonic) [inline ocis-pkg/config/config.go:76]: we do the `-` to `_` change for all yaml tags in a separate PR?

**Comment by butonic** [comment]
Also ... we should probably align the naming in reva in the future

File: .vscode/launch.json (hunk 2 above)
Comment (wkloucek) [reply .vscode/launch.json:44]: yes, seems quite outdated

File: .drone.star (hunk 1 above)
Comment (wkloucek) [reply .drone.star:1724]: actually it's not used for the parallel deployment. So it doesn't really matter

File: extensions/app-provider/pkg/config/config.go (hunk 5 above)
Comment (wkloucek) [reply extensions/app-provider/pkg/config/config.go:71]: this is not about oCIS using invalid certificates, but instead the CS3org WOPI server. Therefore we shouln't use `OCIS_INSECURE` here

File: ocis-pkg/config/config.go (hunk 6 above)
Comment (wkloucek) [reply ocis-pkg/config/config.go:76]: would be fine for me

**Review by dragonchaser — APPROVED** [review]
lgtm

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-5</strong>](https://drone.owncloud.com/owncloud/ocis/11338/45/1) failed. Further test are cancelled...

**Comment by mmattel** [comment]
Info: I have renamed the extension in the admin docs already to this new name. Will be on staging soon.

**Pull request merged**